tests/testthat/_snaps/eval-relocate.md

```diff
@@ -0,0 +1,109 @@
+      after_arg = ".after")
+    Condition
+      Error in `relocate_loc()`:
+      ! Can't supply both `.before` and `.after`.
+
+# can't relocate with out-of-bounds variables by default
+
+    Code
+      (expect_error(relocate_loc(x, c)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
+    Code
+      (expect_error(relocate_loc(x, a, before = c)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
+    Code
+      (expect_error(relocate_loc(x, a, after = c)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
+
+# can relocate with out-of-bounds variables in `expr` if `strict = FALSE`
+
+    Code
+      (expect_error(relocate_loc(x, a, before = c, strict = FALSE)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
+    Code
+      (expect_error(relocate_loc(x, a, after = c, strict = FALSE)))
+    Output
+      <error/vctrs_error_subscript_oob>
+      Error in `relocate_loc()`:
+      ! Can't subset columns that don't exist.
+      x Column `c` doesn't exist.
+
+# can forbid rename syntax
+
+    Code
+      (expect_error(relocate_loc(x, c(foo = b), allow_rename = FALSE)))
+    Output
+      <error/rlang_error>
+      Error in `relocate_loc()`:
+      ! Can't rename variables in this context.
+    Code
+      (expect_error(relocate_loc(x, c(b, foo = b), allow_rename = FALSE)))
+    Output
+      <error/rlang_error>
+      Error in `relocate_loc()`:
+      ! Can't rename variables in this context.
+
+# can forbid empty selections
+
+    Code
+      (expect_error(relocate_loc(x, allow_empty = FALSE)))
+    Output
+      <error/rlang_error>
+      Error in `relocate_loc()`:
+      ! Must select at least one item.
+    Code
+      (expect_error(relocate_loc(mtcars, integer(), allow_empty = FALSE)))
+    Output
+      <error/rlang_error>
+      Error in `relocate_loc()`:
+      ! Must select at least one item.
+    Code
+      (expect_error(relocate_loc(mtcars, starts_with("z"), allow_empty = FALSE)))
+    Output
+      <error/rlang_error>
+      Error in `relocate_loc()`:
+      ! Must select at least one item.
+
+# `before` and `after` forbid renaming
+
+    Code
+      (expect_error(relocate_loc(x, b, before = c(new = c))))
+    Output
+      <error/rlang_error>
+      Error in `relocate_loc()`:
+      ! Can't rename variables in this context.
+    Code
+      (expect_error(relocate_loc(x, b, after = c(new = c))))
+    Output
+      <error/rlang_error>
+      Error in `relocate_loc()`:
+      ! Can't rename variables in this context.
```

Comment:
Another place it would be nice to be able to use `before_arg` and `after_arg`

Reply:
Should we throw a classed error on rename and rethrow it from `relocate()` with a custom message?

Reply:
I think `eval_relocate()` might actually be the one to rethrow it.

It could catch the classed rename error in the two calls to `eval_select()` for `before` and `after` and rethrow a slightly more precise error that uses `before_arg` and `after_arg`.